src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,176 @@
+            time_array_id_columns += ["time_zone"]
+
+        map_table_schema = TableSchema(
+            name="map_table" + str(int(time.time())),
+            time_config=self.to_schema.time_config,
+            time_array_id_columns=time_array_id_columns,
+            value_column="hour",  # this is a workaround
+        )
+        with self._engine.connect() as conn:
+            write_database(dfm, conn, map_table_schema)
+            conn.commit()
+        self._metadata.reflect(self._engine, views=True)
+
+        # Apply mapping and downselect to final cols
+        self._apply_mapping(map_table_schema)
+
+        # TODO write output to parquet?
+
+    def _create_mapping_dataframe_tz_naive(
+        self, dft: pd.DataFrame, to_time_col: str
+    ) -> pd.DataFrame:
+        """Create tz-naive time mapping dataframe"""
+        dfm = dft.copy()
+        dfm["month"] = dfm[to_time_col].dt.month
+        dow = dfm[to_time_col].dt.day_of_week
+        if self.weekday_column == "day_of_week":
+            dfm[self.weekday_column] = dow
+        elif self.weekday_column == "is_weekday":
+            dfm[self.weekday_column] = False
+            dfm.loc[dow < 5, self.weekday_column] = True
+        else:
+            msg = f"No representative time mapping support for time columns: {self.from_schema.time_config.list_time_columns()}"
+            raise NotImplementedError(msg)
+
+        dfm["hour"] = dfm[to_time_col].dt.hour
+        return dfm
+
+    def _create_mapping_dataframe_tz_aware(
+        self, dft: pd.DataFrame, to_time_col: str
+    ) -> pd.DataFrame:
+        """Create tz-aware time mapping dataframe according to to_schema.time_config"""
+        with self._engine.connect() as conn:
+            table = Table(self.from_schema.name, self._metadata)
+            stmt = select(table.c["time_zone"]).distinct().where(table.c["time_zone"].is_not(None))
+            df_tz = read_database(stmt, conn, self.from_schema)
+
+        dfm = []
+        for idx, row in df_tz.iterrows():
+            dfgt = dft.copy()
+            dfgt["timestamp_tmp"] = dfgt[to_time_col].dt.tz_convert(row["time_zone"])
+            dfgt["month"] = dfgt["timestamp_tmp"].dt.month
+
+            dow = dfgt["timestamp_tmp"].dt.day_of_week
+            if self.weekday_column == "day_of_week":
+                dfgt[self.weekday_column] = dow
+            elif self.weekday_column == "is_weekday":
+                dfgt[self.weekday_column] = False
+                dfgt.loc[dow < 5, self.weekday_column] = True
+            else:
+                msg = f"No representative time mapping support for time columns: {self.from_schema.time_config.list_time_columns()}"
+                raise NotImplementedError(msg)
+
+            dfgt["hour"] = dfgt["timestamp_tmp"].dt.hour
+            dfgt["time_zone"] = row["time_zone"]
+            dfm.append(dfgt.drop(columns=["timestamp_tmp"]))
+        dfm = pd.concat(dfm, axis=0, ignore_index=True)
```

Comment:
axis=0 means append the dataframes row-wise (as opposed to column-wise). Ignore_index means ignore that each input df in dfm list has index 0-8759 and instead append them continuously (so for the second df, it'll go from 8760-17519). This would have the same effect as:
`pd.concat(dfm, axis=0).reset_index(drop=True)`